dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -807,6 +809,59 @@ public void createDirectory(String path, CreateDirectoryPOptions options)
     }
   }
 
+  @Override
+  public boolean exists(String path, ExistsPOptions options) {
+    long syncIntervalMs = options.hasCommonOptions()
+        ? (options.getCommonOptions().hasSyncIntervalMs()
+        ? options.getCommonOptions().getSyncIntervalMs() : -1) :
+        -1;
+
+    Optional<DoraMeta.FileStatus> status = mMetaManager.getFromMetaStore(path);
+    if (status.isPresent()) { // metadata exists.
+      if (syncIntervalMs >= 0) {
+        if (System.nanoTime() - status.get().getTs() < syncIntervalMs * Constants.MS_NANO) {
+          // metadata exists and is not out-of-date.
+          return true;
+        }
+      } else {
+        // metadata exists and client does not require to check cache validity.
+        return true;
+      }
+    }
+
+    // TODO(Yimin) Load/reload metadata from UFS
+    try {
+      return mUfs.exists(path);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void setAttribute(String path, SetAttributePOptions options) {
+    try {
+      if (options.hasMode()) {
+        mUfs.setMode(path, ModeUtils.protoToShort(options.getMode()));
+      }
+      if (options.hasOwner() && options.hasGroup()) {
+        mUfs.setOwner(path, options.getOwner(), options.getGroup());
+      } else if (options.hasOwner()) {
+        mUfs.setOwner(path, options.getOwner(), null);
+      } else if (options.hasGroup()) {
+        mUfs.setOwner(path, null, options.getOwner());
+      }
```

Comment:
invalidate metadata cache as the attributes have changed.

Reply:
Yes. I added TODO here, and  @elega will do that.